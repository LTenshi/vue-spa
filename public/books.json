[
  {
    "id": "0",
    "coverUrl": "https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fcdn2.penguin.com.au%2Fcovers%2Foriginal%2F9780141956794.jpg&f=1&nofb=1",
    "name": "In Search of Lost Time",
    "author": "Marcel Proust",
    "year": "1913",
    "rating": "9.8",
    "links": [
      { "name": "Amazon", "url": "https://www.amazon.co.uk" },
      { "name": "iBooks", "url": "https://www.apple.com/uk/apple-books/" },
      { "name": "Play Store", "url": "https://play.google.com/store/books" }
    ],
    "details": "**In Search of Lost Time** (French: Ã€ la recherche du temps perdu), first translated into English as Remembrance of Things Past, and sometimes referred to in French as La Recherche (The Search), is a novel in seven volumes by French author Marcel Proust.This early 20th-century work is his most prominent, known both for its length and its theme of involuntary memory."
  },
  {
    "id": "1",
    "coverUrl": "https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fuploads7.wikiart.org%2Fimages%2Fhonore-daumier%2Fdon-quixote-and-sancho-panza-going-to-the-wedding-gamaches.jpg&f=1&nofb=1",
    "name": "Don Quixote",
    "author": "Miguel de Cervantes",
    "year": "1605",
    "rating": "9.7",
    "links": [
      { "name": "Amazon", "url": "https://www.amazon.co.uk" }
    ],
    "details": "**Don Quixote** is a Spanish epic novel by Miguel de Cervantes. Originally published in two parts, in 1605 and 1615, its full title is The Ingenious Gentleman Don Quixote of La Mancha or, in Spanish, El ingenioso hidalgo don Quijote de la Mancha."
  },
  {
    "id": "2",
    "coverUrl": "https://external-content.duckduckgo.com/iu/?u=http%3A%2F%2Fblogs.harvard.edu%2Fhoughtonmodern%2Ffiles%2F2018%2F09%2FBlog-Ulysses-Cover.jpg&f=1&nofb=1",
    "name": "Ulysses",
    "author": "James Joyce",
    "year": "1922",
    "rating": "9.5",
    "links": [
      { "name": "Amazon", "url": "https://www.amazon.co.uk" },
      { "name": "iBooks", "url": "https://www.apple.com/uk/apple-books/" },
      { "name": "Play Store", "url": "https://play.google.com/store/books" }
    ],
    "details": "**Ulysses** is a modernist novel by Irish writer James Joyce. It was first serialized in parts in the American journal The Little Review from March 1918 to December 1920 and then published in its entirety in Paris by Sylvia Beach on 2 February 1922, Joyce's 40th birthday. It is considered one of the most important works of modernist literature and has been called \"a demonstration and summation of the entire movement.\""
  },
  {
    "id": "3",
    "coverUrl": "https://upload.wikimedia.org/wikipedia/commons/thumb/7/7a/The_Great_Gatsby_Cover_1925_Retouched.jpg/330px-The_Great_Gatsby_Cover_1925_Retouched.jpg",
    "name": "The Great Gatsby",
    "author": "F.Scott Fitzgerald",
    "year": "1925",
    "rating": "9.4",
    "links": [
      { "name": "Amazon", "url": "https://www.amazon.co.uk" },
      { "name": "iBooks", "url": "https://www.apple.com/uk/apple-books/" }
    ],
    "details": "**The Great Gatsby** is a 1925 novel by American writer F. Scott Fitzgerald. Set in the Jazz Age on Long Island, near New York City, the novel depicts first-person narrator Nick Carraway's interactions with mysterious millionaire Jay Gatsby and Gatsby's obsession to reunite with his former lover, Daisy Buchanan."
  },
  {
    "id": "4",
    "coverUrl": "https://upload.wikimedia.org/wikipedia/commons/thumb/3/36/Moby-Dick_FE_title_page.jpg/330px-Moby-Dick_FE_title_page.jpg",
    "name": "Moby Dick",
    "author": "Herman Melville",
    "year": "1851",
    "rating": "9.3",
    "links": [
      { "name": "Amazon", "url": "https://www.amazon.co.uk" }
    ],
    "details": "**Don Quixote** is a Spanish epic novel by Miguel de Cervantes. Originally published in two parts, in 1605 and 1615, its full title is The Ingenious Gentleman Don Quixote of La Mancha or, in Spanish, El ingenioso hidalgo don Quijote de la Mancha."
  },
  {
    "id": "5",
    "coverUrl": "https://upload.wikimedia.org/wikipedia/en/0/08/Leviathan_Wakes.jpg",
    "name": "Leviathan Wakes",
    "author": "James S.A. Corey",
    "year": "2011",
    "rating": "9.2",
    "links": [
      { "name": "Amazon", "url": "https://www.amazon.co.uk" }
    ],
    "details": "**Leviathan Wakes** is a science fiction novel by James S. A. Corey, the pen name of American writers Daniel Abraham and Ty Franck. It is the first book in the Expanse series, followed by Caliban's War (2012), Abaddon's Gate (2013) and six other novels. Leviathan Wakes was nominated for the 2012 Hugo Award for Best Novel and the 2012 Locus Award for Best Science Fiction Novel. The novel was adapted for television in 2015 as the first season-and-a-half of The Expanse by Syfy. Five short stories that take place before, during, or after Leviathan Wakes were published between 2011 and 2019."
  },

  {
    "id": "6",
    "coverUrl": "https://upload.wikimedia.org/wikipedia/en/1/15/Calibans_War.jpg",
    "name": "Caliban's War",
    "author": "James S.A. Corey",
    "year": "2012",
    "rating": "9.1",
    "links": [
      { "name": "Amazon", "url": "https://www.amazon.co.uk" }
    ],
    "details": "**Caliban's War** is a 2012 science fiction novel by James S. A. Corey (pen name of Daniel Abraham and Ty Franck). It is about a conflict in the Solar System that involves Earth, Mars, and the Asteroid Belt (colonies of people living on asteroids, referred to as \"Belters\")."
  },
  {
    "id": "7",
    "coverUrl": "https://upload.wikimedia.org/wikipedia/en/6/6b/James_S.A._Corey_-_Abaddon%27s_Gate.jpeg",
    "name": "Abaddon Gate",
    "author": "James S.A. Corey",
    "year": "2013",
    "rating": "9.0",
    "links": [
      { "name": "Amazon", "url": "https://www.amazon.co.uk" }
    ],
    "details": "**Abaddon's Gate** is a science fiction novel by James S. A. Corey (pen name of Daniel Abraham and Ty Franck). It is about a conflict in the Solar System that involves Earth, Mars, and the Asteroid Belt (colonies of people living on asteroids, referred to as \"Belters\")."
  },
  {
    "id": "8",
    "coverUrl": "https://upload.wikimedia.org/wikipedia/en/0/08/Leviathan_Wakes.jpg",
    "name": "Leviathan Wakes",
    "author": "James S.A. Corey",
    "year": "2011",
    "rating": "9.2",
    "links": [
      { "name": "Amazon", "url": "https://www.amazon.co.uk" }
    ],
    "details": "**Leviathan Wakes** is a science fiction novel by James S. A. Corey, the pen name of American writers Daniel Abraham and Ty Franck. It is the first book in the Expanse series, followed by Caliban's War (2012), Abaddon's Gate (2013) and six other novels. Leviathan Wakes was nominated for the 2012 Hugo Award for Best Novel and the 2012 Locus Award for Best Science Fiction Novel. The novel was adapted for television in 2015 as the first season-and-a-half of The Expanse by Syfy. Five short stories that take place before, during, or after Leviathan Wakes were published between 2011 and 2019."
  },
  {
    "id": "9",
    "coverUrl": "https://upload.wikimedia.org/wikipedia/en/b/be/Cibola_Burn.jpg",
    "name": "Cibola Burn",
    "author": "James S.A. Corey",
    "year": "2014",
    "rating": "8.9",
    "links": [
      { "name": "Amazon", "url": "https://www.amazon.co.uk" }
    ],
    "details": "**Cibola Burn** is a 2014 science fiction novel by James S. A. Corey (pen name of Daniel Abraham and Ty Franck) and the fourth book in The Expanse series. It follows the crew of the Rocinante as they join the flood of humanity out into the galaxy, using the gates built by the ancient civilization that also produced the protomolecule."
  },
  {
    "id": "10",
    "coverUrl": "https://upload.wikimedia.org/wikipedia/en/6/60/Nemesis_Games.jpg",
    "name": "Nemesis Games",
    "author": "James S.A. Corey",
    "year": "2015",
    "rating": "8.8",
    "links": [
      { "name": "Amazon", "url": "https://www.amazon.co.uk" }
    ],
    "details": "**Nemesis Games** is a 2015 science fiction novel by James S. A. Corey, the pen name of Daniel Abraham and Ty Franck, and the fifth book in their The Expanse series. It is the sequel to Cibola Burn. "
  },
  {
    "id": "11",
    "coverUrl": "https://upload.wikimedia.org/wikipedia/en/6/68/Babylons_Ashes.jpg",
    "name": "Babylon's Ashes",
    "author": "James S.A. Corey",
    "year": "2011",
    "rating": "8.7",
    "links": [
      { "name": "Amazon", "url": "https://www.amazon.co.uk" }
    ],
    "details": "**Babylon's Ashes** is a science fiction novel by James S. A. Corey, the pen name of Daniel Abraham and Ty Franck, and the sixth book in their The Expanse series."
  }
]