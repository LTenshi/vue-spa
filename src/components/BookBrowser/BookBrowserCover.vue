<template>
  <tr>
    <th class="imageHeader">
        <img :src="bookCover" @click="openDetails" class="imageContainer">
    </th>
    <th class="bookName">
        <div>{{bookName}}</div>
        <div class="bookAuthor">{{bookAuthor}}</div>
    </th>
  </tr>
</template>

<script>
export default {
    props: {
        bookName: {
            type: String,
            required: true,
        },
        bookAuthor: {
            type: String,
            required: true,
        },
        bookCover: {
            type: String,
            required: true,
        },
    },
    methods: {
        openDetails() {
            this.$emit("openDetails");
        }
    }
}
</script>

<style>
.bookName, .bookAuthor {
    font-size: 13pt;
}

.bookName {
    vertical-align: top;
    padding: 20px;
    text-align: left;
}

.bookAuthor {
    font-weight: 400;
    color: rgb(175, 175, 175);
}

.imageContainer {
    max-width: 140px;
}
</style>