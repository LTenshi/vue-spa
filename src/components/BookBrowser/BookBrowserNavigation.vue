<template>
    <th class="container" colspan="4">
        <button @click="moveBack" :style="isBackPossible" :disabled="isBackPossible">&#10094;</button>
        Page {{currPage}} of {{maxPages}} 
        <button @click="moveForward" :style="isForwardPossible" :disabled="isForwardPossible">&#10095;</button>
    </th>
</template>

<script>
export default {
 props: {
    maxPages: {
      type: Number,
      required: true
    },
    currPage: {
        type: Number,
        required: true,
    }
  },
  methods: {
    moveBack() {
        this.$emit("navigationEvent", this.currPage - 1);
    },
    moveForward() {
        this.$emit("navigationEvent", this.currPage + 1);
    }
  },
  computed: {
    isBackPossible() {
        return this.currPage === 1 ? { color: "lightgrey", cursor: "unset" } : false
    },
    isForwardPossible() {
        return this.currPage === this.maxPages ? { color: "lightgrey", cursor: "unset" } : false
    }
  }
}
</script>

<style>
.container {
    width: 100%;
}

button {
	background: none;
    color: grey;
	border: none;
	font: inherit;
	cursor: pointer;
}
</style>