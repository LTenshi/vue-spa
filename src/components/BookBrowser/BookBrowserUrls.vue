<template>
  <th>
    <a v-for="link in bookLinks" :key="link.name" :href="link.url" class="bookLink">{{link.name}}<br></a>
  </th>
</template>

<script>
export default {
  props: {
    bookLinks: {
      type: Array,
      default: () => []
    }
  }
}
</script>

<style>
.bookLink {
  color: rgb(136, 221, 0);
  text-decoration: none;
}

.bookLink > a:link {
  text-decoration: none;
}

.bookLink > a:visited {
  text-decoration: none;
}

.bookLink > a:hover {
  text-decoration: none;
}

.bookLink > a:active {
  text-decoration: none;
}
</style>