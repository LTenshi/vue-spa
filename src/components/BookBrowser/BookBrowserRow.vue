<template>
  <tr>
    <book-browser-cover 
        :bookCover="bookCover" 
        :bookName="bookName" 
        :bookAuthor="bookAuthor"
        @openDetails="openDetails"/>
    <th class="fontStyle">
        {{this.bookYear}}
    </th>
    <th class="fontStyle">
        {{this.bookRating}}/10
    </th>
    <book-browser-urls :bookLinks="bookLinks"/>
  </tr>
  <th v-if="detailsExpanded" colspan="4" class="detailsPanel" v-html="this.bookDetails"></th>
</template>

<script>
import BookBrowserCover from './BookBrowserCover.vue';
import BookBrowserUrls from './BookBrowserUrls.vue';
import showdown from "showdown";
export default {
    components: { BookBrowserCover, BookBrowserUrls },
    data() {
        return {
            detailsExpanded: false
        }
    },
    props: {
        book: {
            type: Object,
            required: true,
        },
    },
    computed: {
        bookCover() {
            return this.book.coverUrl;
        },
        bookName() {
            return this.book.name;
        },
        bookId() {
            return this.book.id;
        },
        bookAuthor() {
            return this.book.author;
        },
        bookYear() {
            return this.book.year;
        },
        bookRating() {
            return this.book.rating;
        },
        bookLinks() {
            return this.book.links;
        },
        bookDetails() {
            var converter = new showdown.Converter();
            return converter.makeHtml(this.book.details);
        }
    },
    methods: {
        openDetails() {
            this.detailsExpanded = !this.detailsExpanded;
        }
    }
}
</script>

<style>
.detailsPanel {
    font-weight: 450;
    text-align: initial;
}

.fontStyle {
    font-weight: 450;
}
</style>