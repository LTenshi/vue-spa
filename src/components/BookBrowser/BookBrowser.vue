<template>
  <table id="bookbrowser">
    <thead>
      <h3>Most popular Books of All time</h3>
    </thead>
    <tbody class="rowContainer">
      <tr>
        <th v-for="value in columns" :key="value.title" class="tHeader" :style="headerStyle(value.style)">
          {{capitalise(value.title)}}
        </th>
      </tr>
      <book-browser-row 
        v-for="book in books" 
        :key="book.id"
        :book="book">
      </book-browser-row>
    </tbody>
    <tfoot>
      <book-browser-navigation :maxPages="maxPages" :currPage="currPage" @navigationEvent="navigationEventHandler"/>
    </tfoot>
  </table>
</template>

<script>
import BookBrowserRow from './BookBrowserRow.vue'
import BookBrowserNavigation from "./BookBrowserNavigation.vue"
export default {
  components: { BookBrowserRow, BookBrowserNavigation },
  data: function() {
    return {
      columns: [
        { title: "Title", style: { width: "60", textAlign: "initial" } },
        { title: "Published", style: { width: "10" } },
        { title: "Rating", style:{ width: "10" } },
        { title: "Buy On", style: { width: "10" } },
      ]
    }
  },
  props: {
    books: {
      type: Array,
      required: true,
      default: () => []
    },
    maxPages: {
      type: Number,
      required: true
    },
    currPage: {
      type: Number,
      required: true
    }
  },
  methods: {
    capitalise(value) {
      return value.toUpperCase();
    },
    headerStyle(style) {
      let styleReturn = {};
      styleReturn.width = style.width + "%";
      if (style.textAlign) {
        styleReturn.textAlign = style.textAlign;
      }
      return styleReturn
    },
    navigationEventHandler(navEvent) {
      this.$emit("navigationEvent", navEvent);
    }
  },
}
</script>

<style>

#bookbrowser {
  width: 100%;
  background-color: white;
  border-radius: 5px;
  padding: 10px;
}

.rowContainer > tr:nth-child(even) {
  background-color: rgb(245, 245, 245);
}

.tHeader {
  color: rgb(175, 175, 175);
}

</style>