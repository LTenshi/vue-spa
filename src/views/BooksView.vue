<template>
  <div>
    <h1>Books</h1>
    <book-browser :books="books" :maxPages="pages" :currPage="currPage" @navigationEvent="handlePageChange"/>
  </div>
</template>

<script>
import BookBrowser from '@/components/BookBrowser/BookBrowser.vue'
// import BooksJSON from "../assets/books.json"
export default {
  components: { BookBrowser },
  data() {
    return {
      books: [],
      itemsToLoad: 5,
      pages: 0,
      currPage: 1,
    }
  },
  methods: {
    //startIndex and endIndex will only simulate the pagination received from API.
    //This function will always load all books.
    async fetchBooks() {
      //Can be replaced with any other url.
      let response = await fetch("books.json");
      let json = await response.json();

      //Pagination below
      let index = 0 + ( (this.currPage - 1) * this.itemsToLoad);
      this.books = json.slice(0 + ( (this.currPage - 1) * this.itemsToLoad), index + this.itemsToLoad);
      this.pages = Math.ceil(json.length / this.itemsToLoad);
    },
    handlePageChange(payload) {
      this.currPage = payload;
      this.fetchBooks();
    }
  },
  mounted() {
    this.fetchBooks();
  }
}
</script>

<style>

</style>