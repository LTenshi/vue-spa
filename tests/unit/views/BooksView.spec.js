import BooksView from "../../../src/views/BooksView";
import BookBrowser from "../../../src/components/BookBrowser/BookBrowser";
import BookBrowserNavigation from "../../../src/components/BookBrowser/BookBrowserNavigation";
import books from "../../../public/books"
import { mount } from "@vue/test-utils";
import { jest } from '@jest/globals';



describe("Books View", () => {
    let wrapper;
    let BookBrowserComponent;
    let BookBrowserNavigationComponent;

    beforeEach(() => {
        //Fetch is not a node function yet. Need to mock it out.
        global.fetch = jest.fn(() =>
            Promise.resolve({
                json: () => Promise.resolve(books),
            })
        );
        wrapper = mount(BooksView);
        BookBrowserComponent = wrapper.findComponent(BookBrowser);
        BookBrowserNavigationComponent = BookBrowserComponent.findComponent(BookBrowserNavigation);

    })

    afterEach(() => {
        global.fetch.mockRestore();
    })

    it("Should match the snapshot", async () => {
        expect(wrapper.element).toMatchSnapshot();

        await BookBrowserNavigationComponent.vm.moveForward();
        await BookBrowserNavigationComponent.vm.moveForward();

        expect(wrapper.element).toMatchSnapshot("2 moved forward");
    })

    it('Should contain an array of books', () => {
        expect(wrapper.vm.books.length).toBeGreaterThan(0);
    });

    it('Should contain an array of 5 books', () => {
        expect(wrapper.vm.books.length).toBe(5);
    });

    it('Should call fetch', () => {
        expect(global.fetch.mock.calls.length).toBe(1);
    });

    it('Should have 3 pages', () => {
        expect(wrapper.vm.pages).toBe(3);
    });

    it('Should render the book browser', () => {
        expect(BookBrowserComponent.exists()).toBe(true);
    });

    it('Book browser should have books, pages and a current page', () => {
        expect(BookBrowserComponent.vm.books.length).toBe(5);
        expect(BookBrowserComponent.vm.maxPages).toBe(3);
        expect(BookBrowserComponent.vm.currPage).toBe(1);
    });

    it('Should show 2 books after navigating twice', async () => {
        await BookBrowserNavigationComponent.vm.moveForward();
        await BookBrowserNavigationComponent.vm.moveForward();

        expect(BookBrowserComponent.vm.currPage).toBe(3);
        expect(BookBrowserComponent.vm.books.length).toBe(2);
        expect(BookBrowserComponent.vm.maxPages).toBe(3);
    });
    
});