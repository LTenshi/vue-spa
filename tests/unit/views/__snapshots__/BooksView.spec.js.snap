// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Books View Should match the snapshot 1`] = `
<div>
  <h1>
    Books
  </h1>
  <table
    id="bookbrowser"
  >
    <thead>
      <h3>
        Most popular Books of All time
      </h3>
    </thead>
    <tbody
      class="rowContainer"
    >
      <tr>
        
        <th
          class="tHeader"
          style="width: 60%; text-align: initial;"
        >
          TITLE
        </th>
        <th
          class="tHeader"
          style="width: 10%;"
        >
          PUBLISHED
        </th>
        <th
          class="tHeader"
          style="width: 10%;"
        >
          RATING
        </th>
        <th
          class="tHeader"
          style="width: 10%;"
        >
          BUY ON
        </th>
        
      </tr>
      
      
      <tr>
        <tr>
          <th
            class="imageHeader"
          >
            <img
              class="imageContainer"
              src="https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fcdn2.penguin.com.au%2Fcovers%2Foriginal%2F9780141956794.jpg&f=1&nofb=1"
            />
          </th>
          <th
            class="bookName"
          >
            <div>
              In Search of Lost Time
            </div>
            <div
              class="bookAuthor"
            >
              Marcel Proust
            </div>
          </th>
        </tr>
        <th
          class="fontStyle"
        >
          1913
        </th>
        <th
          class="fontStyle"
        >
          9.8/10 
        </th>
        <th>
          
          <a
            class="bookLink"
            href="https://www.amazon.co.uk"
          >
            Amazon
            <br />
          </a>
          <a
            class="bookLink"
            href="https://www.apple.com/uk/apple-books/"
          >
            iBooks
            <br />
          </a>
          <a
            class="bookLink"
            href="https://play.google.com/store/books"
          >
            Play Store
            <br />
          </a>
          
        </th>
      </tr>
      <!--v-if-->
      
      
      <tr>
        <tr>
          <th
            class="imageHeader"
          >
            <img
              class="imageContainer"
              src="https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fuploads7.wikiart.org%2Fimages%2Fhonore-daumier%2Fdon-quixote-and-sancho-panza-going-to-the-wedding-gamaches.jpg&f=1&nofb=1"
            />
          </th>
          <th
            class="bookName"
          >
            <div>
              Don Quixote
            </div>
            <div
              class="bookAuthor"
            >
              Miguel de Cervantes
            </div>
          </th>
        </tr>
        <th
          class="fontStyle"
        >
          1605
        </th>
        <th
          class="fontStyle"
        >
          9.7/10 
        </th>
        <th>
          
          <a
            class="bookLink"
            href="https://www.amazon.co.uk"
          >
            Amazon
            <br />
          </a>
          
        </th>
      </tr>
      <!--v-if-->
      
      
      <tr>
        <tr>
          <th
            class="imageHeader"
          >
            <img
              class="imageContainer"
              src="https://external-content.duckduckgo.com/iu/?u=http%3A%2F%2Fblogs.harvard.edu%2Fhoughtonmodern%2Ffiles%2F2018%2F09%2FBlog-Ulysses-Cover.jpg&f=1&nofb=1"
            />
          </th>
          <th
            class="bookName"
          >
            <div>
              Ulysses
            </div>
            <div
              class="bookAuthor"
            >
              James Joyce
            </div>
          </th>
        </tr>
        <th
          class="fontStyle"
        >
          1922
        </th>
        <th
          class="fontStyle"
        >
          9.5/10 
        </th>
        <th>
          
          <a
            class="bookLink"
            href="https://www.amazon.co.uk"
          >
            Amazon
            <br />
          </a>
          <a
            class="bookLink"
            href="https://www.apple.com/uk/apple-books/"
          >
            iBooks
            <br />
          </a>
          <a
            class="bookLink"
            href="https://play.google.com/store/books"
          >
            Play Store
            <br />
          </a>
          
        </th>
      </tr>
      <!--v-if-->
      
      
      <tr>
        <tr>
          <th
            class="imageHeader"
          >
            <img
              class="imageContainer"
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/7/7a/The_Great_Gatsby_Cover_1925_Retouched.jpg/330px-The_Great_Gatsby_Cover_1925_Retouched.jpg"
            />
          </th>
          <th
            class="bookName"
          >
            <div>
              The Great Gatsby
            </div>
            <div
              class="bookAuthor"
            >
              F.Scott Fitzgerald
            </div>
          </th>
        </tr>
        <th
          class="fontStyle"
        >
          1925
        </th>
        <th
          class="fontStyle"
        >
          9.4/10 
        </th>
        <th>
          
          <a
            class="bookLink"
            href="https://www.amazon.co.uk"
          >
            Amazon
            <br />
          </a>
          <a
            class="bookLink"
            href="https://www.apple.com/uk/apple-books/"
          >
            iBooks
            <br />
          </a>
          
        </th>
      </tr>
      <!--v-if-->
      
      
      <tr>
        <tr>
          <th
            class="imageHeader"
          >
            <img
              class="imageContainer"
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/3/36/Moby-Dick_FE_title_page.jpg/330px-Moby-Dick_FE_title_page.jpg"
            />
          </th>
          <th
            class="bookName"
          >
            <div>
              Moby Dick
            </div>
            <div
              class="bookAuthor"
            >
              Herman Melville
            </div>
          </th>
        </tr>
        <th
          class="fontStyle"
        >
          1851
        </th>
        <th
          class="fontStyle"
        >
          9.3/10 
        </th>
        <th>
          
          <a
            class="bookLink"
            href="https://www.amazon.co.uk"
          >
            Amazon
            <br />
          </a>
          
        </th>
      </tr>
      <!--v-if-->
      
      
    </tbody>
    <tfoot>
      <th
        class="container"
        colspan="4"
      >
        <button
          disabled=""
          style="color: lightgrey; cursor: unset;"
        >
          ❮
        </button>
         Page 1 of 3 
        <button>
          ❯
        </button>
      </th>
    </tfoot>
  </table>
</div>
`;

exports[`Books View Should match the snapshot: 2 moved forward 1`] = `
<div>
  <h1>
    Books
  </h1>
  <table
    id="bookbrowser"
  >
    <thead>
      <h3>
        Most popular Books of All time
      </h3>
    </thead>
    <tbody
      class="rowContainer"
    >
      <tr>
        
        <th
          class="tHeader"
          style="width: 60%; text-align: initial;"
        >
          TITLE
        </th>
        <th
          class="tHeader"
          style="width: 10%;"
        >
          PUBLISHED
        </th>
        <th
          class="tHeader"
          style="width: 10%;"
        >
          RATING
        </th>
        <th
          class="tHeader"
          style="width: 10%;"
        >
          BUY ON
        </th>
        
      </tr>
      
      
      <tr>
        <tr>
          <th
            class="imageHeader"
          >
            <img
              class="imageContainer"
              src="https://upload.wikimedia.org/wikipedia/en/6/60/Nemesis_Games.jpg"
            />
          </th>
          <th
            class="bookName"
          >
            <div>
              Nemesis Games
            </div>
            <div
              class="bookAuthor"
            >
              James S.A. Corey
            </div>
          </th>
        </tr>
        <th
          class="fontStyle"
        >
          2015
        </th>
        <th
          class="fontStyle"
        >
          8.8/10 
        </th>
        <th>
          
          <a
            class="bookLink"
            href="https://www.amazon.co.uk"
          >
            Amazon
            <br />
          </a>
          
        </th>
      </tr>
      <!--v-if-->
      
      
      <tr>
        <tr>
          <th
            class="imageHeader"
          >
            <img
              class="imageContainer"
              src="https://upload.wikimedia.org/wikipedia/en/6/68/Babylons_Ashes.jpg"
            />
          </th>
          <th
            class="bookName"
          >
            <div>
              Babylon's Ashes
            </div>
            <div
              class="bookAuthor"
            >
              James S.A. Corey
            </div>
          </th>
        </tr>
        <th
          class="fontStyle"
        >
          2011
        </th>
        <th
          class="fontStyle"
        >
          8.7/10 
        </th>
        <th>
          
          <a
            class="bookLink"
            href="https://www.amazon.co.uk"
          >
            Amazon
            <br />
          </a>
          
        </th>
      </tr>
      <!--v-if-->
      
      
    </tbody>
    <tfoot>
      <th
        class="container"
        colspan="4"
      >
        <button>
          ❮
        </button>
         Page 3 of 3 
        <button
          disabled=""
          style="color: lightgrey; cursor: unset;"
        >
          ❯
        </button>
      </th>
    </tfoot>
  </table>
</div>
`;
